### PR TITLE
Add Copilot custom instructions for code review and cloud agent

### PR DESCRIPTION
**Pull request by italo-sampaio**
## Description
Add `.github/copilot-instructions.md`, a repository-wide custom
instructions file for GitHub Copilot. The file targets the two
consumers documented in GitHub's official guide: Copilot code review
(runs on every pull request) and Copilot cloud agent (authors pull
requests). It captures, in roughly two pages:

- A short repo summary (Java 17, Gradle 8.6, `rskj-core`, entry point
  `co.rsk.Start`).
- The team's PR review priorities: review-ability first, Boy Scout
  Rule, separate commits for cosmetic vs core changes, incremental
  refactors, anti-flakiness rules, dependency hygiene, PR-template
  compliance.
- The PR-blocking CI workflows and the exact gradle commands each
  one runs.
- Build, test, and lint commands that mirror CI.
- The project layout, with consensus-critical packages called out.
- A list of sensitive areas requiring extra reviewer scrutiny.

The file opens and closes with a "trust these instructions" notice so
Copilot does not waste effort re-exploring the repo on each PR.

## Motivation and Context
Copilot code review currently sees this repository without any
project context. It cannot reliably predict which CI gates will run,
which areas are consensus-critical, or what the team's review
conventions are. The result is review feedback that is generic at
best and occasionally misleading. This file gives Copilot a durable,
repo-resident reference and brings its PR-review feedback in line
with the team's actual standards.

## How Has This Been Tested?
Documentation-only change; no code paths are exercised.

Static validations performed locally:
- Word count is under the two-page limit suggested by the official
  guide.
- Every file path referenced in the document exists at HEAD.
- Every gradle command referenced matches what the corresponding
  workflow under `.github/workflows/` actually runs.

The instructions will take effect on the next pull request opened
against this repository after merge.

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**: This file is consumed automatically by
  Copilot code review on every pull request and by Copilot cloud
  agent when it authors PRs. The companion path-specific format
  (`.github/instructions/*.instructions.md`) is intentionally
  deferred to a future change; this single file applies to both
  Copilot tools.

### CODE REVIEW

File: .github/copilot-instructions.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+- **Test coverage cannot regress.** New code paths — including edge cases and error paths — require unit tests. Untested exception handling needs explicit written justification in the PR description.
+- **Anti-flakiness.** This repository has a known flakiness backlog the team is actively reducing. Flag new or changed tests that use `Thread.sleep` for synchronization, real-clock timing dependencies, real network calls without stubs, parallel-execution ordering assumptions, fragile filesystem assumptions, or that are newly `@Disabled` / `@Ignore`d.
+- **Clean Code is a guideline, not a rule.** The codebase is large and partly legacy; do not demand rewrites in legacy modules. Apply Clean Code principles where reasonable.
+- **No unrelated reordering or renaming.** Diffs that reorder or rename unchanged code without a clear benefit hurt review-ability — flag them.
+- **Dependencies are discouraged.** Any added or upgraded dependency must be accompanied by a matching PR to the `rsksmart/reproducible-builds` repository so the downloaded binary SHA256 can be verified against an independent compile. Flag dependency changes that lack this companion PR.
```
Comment (marcos-iov) [inline .github/copilot-instructions.md]: I don't think this applies to every dependency. reproducible-builds contains reproducibles for our projects, but for external dependencies we won't have a matching PR there

**Review by dlealmac — APPROVED** [review]
LGTM

**Review by marcos-iov — APPROVED** [review]

File: .github/copilot-instructions.md (hunk 1 above)
Comment (italo-sampaio) [reply .github/copilot-instructions.md]: Addressed, thanks!
